google_analytics/aggregation.py

```diff
@@ -92,58 +93,193 @@ def aggregate(df):
+            "pageviews": sum,
+            "transactions": sum,
+            "visits": sum,
+            "Avg. Session Duration": sum,
+            "Bounce Rate": sum,
+            "transactionRevenue": sum,
+            "visitStartTime": "first",
+            "date": "first"
+        }
+
+    def agg_date(s):
+        try:
+            date = pd.datetime.strptime(str(s), '%Y-%m-%d')
+            return "{}_{}".format(date.month, date.year)
+        except ValueError:
+            return "{}_{}".format(s.month, s.year)
 
+    df["date_temp"] = df["date"].apply(agg_date)
     print("First grouping, this will take a while")
-    grouped = df.groupby(["fullVisitorId", "date"], as_index=False).agg(agg)
+    return df.groupby(["fullVisitorId", "date_temp"], as_index=False).agg(agg)
 
-    # Long to wide pivot - one row per visitor.
-    dynamic_columns = ['transactionRevenue', 'visits', 'transactions', 'pageviews']
-    wide = pd.pivot_table(grouped, index='fullVisitorId', columns='date', values=dynamic_columns)
 
-    # Collapse multi-index.
-    wide.columns = wide.columns.to_series().str.join('_')
-    wide.reset_index(inplace=True)
+def deal_static(train, test, selec_top_per=0.5, max_cat=10):
+    """This function merge the train and test and creates
+    OHE for every static variable
 
-    # Now let's also get the static columns.
-    static_columns = list(set(grouped.columns) - set(dynamic_columns) - {"date"})
-    static = grouped[static_columns]
+    Parameters
+    ----------
+    train: DataFrame
+    test: DataFrame
+    selec_top_per: float
+        to select the top percentage per category
+    max_cat: int
+        the maximun number of categories
+    Returns
+    -------
+    DataFrame
+        train and test sets aggregated per user and date with the
+        static variables as OHE
+    """
+    def transform_date(x):
+        try:
+            return pd.datetime.strptime(str(x), '%Y-%m-%d')
+        except ValueError:
+            return x
+
+    merged = pd.concat([train, test], sort=False).reset_index(drop=True)
+    merged['transactions'].fillna(0, inplace=True)
+    merged['transactionRevenue'].fillna(0.0, inplace=True)
+    # Create some features
+    merged['date'] = merged['date'].apply(lambda x: transform_date(x))
+    merged['weekday'] = merged['date'].apply(lambda x: x.weekday())
+    # Aggregate the dataset
+    merged = group_data(merged)
+    # Reduce categories on static columns
+    OHE_reduce = ['operatingSystem', 'country', 'browser']  # noqa
+    merged = reduce_categories(merged, OHE_reduce, selec_top_per, max_cat)
+    # create OHE
+    merged = one_hot_encode_categoricals(merged)
+    return merged
+
+
+def create_train_test(merged, x_train_dates=('2016-08-01', '2017-11-30'), y_train_dates=('2017-12-01', '2018-01-31'),
+                      x_test_dates=('2017-08-01', '2018-11-30'), y_test_dates=None):
+    """Splits the merged dataset on the defined time intervals.
+    """
+    def split_y(y):
```

Comment:
Something goes wrong here, I don't know exactly what. But the 'target' is not the sum of all fullVisitorId's